editor/editor_build_profile.cpp

```diff
@@ -0,0 +1,798 @@
+	"openxr",
+	"opengl3",
+	"vulkan",
+};
+
+const bool EditorBuildProfile::build_option_disable_values[BUILD_OPTION_MAX] = {
+	// This map to SCons build options
+	true,
+	true,
+	true,
+	true,
+	false,
+	false,
+	false
+};
+
+void EditorBuildProfile::set_disable_class(const StringName &p_class, bool p_disabled) {
+	if (p_disabled) {
+		disabled_classes.insert(p_class);
+	} else {
+		disabled_classes.erase(p_class);
+	}
+}
+
+bool EditorBuildProfile::is_class_disabled(const StringName &p_class) const {
+	if (p_class == StringName()) {
+		return false;
+	}
+	return disabled_classes.has(p_class) || is_class_disabled(ClassDB::get_parent_class_nocheck(p_class));
+}
+
+void EditorBuildProfile::set_item_collapsed(const StringName &p_class, bool p_collapsed) {
+	if (p_collapsed) {
+		collapsed_classes.insert(p_class);
+	} else {
+		collapsed_classes.erase(p_class);
+	}
+}
+
+bool EditorBuildProfile::is_item_collapsed(const StringName &p_class) const {
+	return collapsed_classes.has(p_class);
+}
+
+void EditorBuildProfile::set_disable_build_option(BuildOption p_build_option, bool p_disable) {
+	ERR_FAIL_INDEX(p_build_option, BUILD_OPTION_MAX);
+	build_options_disabled[p_build_option] = p_disable;
+}
+
+void EditorBuildProfile::clear_disabled_classes() {
+	disabled_classes.clear();
+	collapsed_classes.clear();
+}
+
+bool EditorBuildProfile::is_build_option_disabled(BuildOption p_build_option) const {
+	ERR_FAIL_INDEX_V(p_build_option, BUILD_OPTION_MAX, false);
+	return build_options_disabled[p_build_option];
+}
+
+bool EditorBuildProfile::get_build_option_disable_value(BuildOption p_build_option) {
+	ERR_FAIL_INDEX_V(p_build_option, BUILD_OPTION_MAX, false);
+	return build_option_disable_values[p_build_option];
+}
+
+void EditorBuildProfile::set_force_detect_classes(const String &p_classes) {
+	force_detect_classes = p_classes;
+}
+
+String EditorBuildProfile::get_force_detect_classes() const {
+	return force_detect_classes;
+}
+
+String EditorBuildProfile::get_build_option_name(BuildOption p_build_option) {
+	ERR_FAIL_INDEX_V(p_build_option, BUILD_OPTION_MAX, String());
+	const char *build_option_names[BUILD_OPTION_MAX] = {
+		TTRC("3D Engine"),
+		TTRC("2D Physics"),
+		TTRC("3D Physics"),
+		TTRC("Navigation"),
+		TTRC("XR"),
+		TTRC("RenderingDevice"),
+		TTRC("OpenGL"),
+		TTRC("Vulkan"),
+	};
+	return TTRGET(build_option_names[p_build_option]);
+}
+
+String EditorBuildProfile::get_build_option_description(BuildOption p_build_option) {
+	ERR_FAIL_INDEX_V(p_build_option, BUILD_OPTION_MAX, String());
+
+	const char *build_option_descriptions[BUILD_OPTION_MAX] = {
+		TTRC("3D Nodes as well as RenderingServer access to 3D features."),
+		TTRC("2D Physics nodes and PhysicsServer2D."),
+		TTRC("3D Physics nodes and PhysicsServer3D."),
+		TTRC("Navigation, both 2D and 3D."),
+		TTRC("XR (AR and VR)."),
+		TTRC("RenderingDevice based rendering (requires OpenGL back-end)."),
+		TTRC("OpenGL back-end (requires RenderingDevice back-end)."),
```

Comment:
That's a bit confusing, I thought OpenGL doesn't use RenderingDevice.
Do you mean to say that to disable RenderingDevice, then OpenGL must be enabled, and vice versa?

Then maybe:
```suggestion
		TTRC("RenderingDevice based rendering (if disabled, the OpenGL back-end is required)."),
		TTRC("OpenGL back-end (if disabled, the RenderingDevice back-end is required)."),
```